core/shared/src/main/scala/org/typelevel/log4cats/extras/WriterTStructuredLogger.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats.extras
+
+import cats.data.WriterT
+import cats.kernel.Monoid
+import cats.syntax.all._
+import cats.{~>, Alternative, Applicative, Foldable, Monad}
+import org.typelevel.log4cats.{SelfAwareStructuredLogger, StructuredLogger}
+
+/**
+ * >>> WARNING READ BEFORE USAGE! <<< This logger will NOT log anything if `F` fails!
+ *
+ * Running the `WriterT` instance will yield a value of type `F[(G[LogMessage], A)]`. As a result,
+ * the logged messages can be materialized if and only `F` succeeds.
+ *
+ * This is particularly important because of the way that `cats.effect.IO` handles cancellation and
+ * timeouts.
```

Comment:
Good reasons not to use it, but no reason to use it.  Maybe highlight that it works without a Sync instance?

Reply:
Better way to look at it: if your log values are not a global operational log, but annotations with referential integrity to an entity, this is may be a good choice.

Reply:
I'll have a think on how to word that

Reply:
Yeah, mine still isn't great.  The key difference, for better and for worse, is that it matches the life cycle of the value.  If the logs are relevant in the absence of the value, this is very bad.  If the logs are tightly coupled to the value and need not be retained longer than the value, this is highly convenient!

Reply:
I gave it a shot. To help avoid having to put this in every class header, I moved the documentation to it's own `README.md`